src/main/java/net/pms/util/SubtitleColor.java

```diff
@@ -0,0 +1,466 @@
+		}
+	}
+
+	/**
+	 * Try to create a {@link SubtitleColor} from {@code subtitleColor}. The
+	 * parsing will try 4 different parsing methods:
+	 *
+	 * <ul>
+	 * <li>First try to parse {@code subtitleColor} as a hexadecimal string.
+	 * Several different notations are supported and it can be with or without
+	 * alpha. Valid forms are RRGGBB or RRGGBBAA with one of the following
+	 * prefixes: "x'", "%", "#", "0x", "\x", "%x", "$", "h'", "16#", "16r",
+	 * "#x", "#16r", "&amp;h" or "0h".</li>
+	 * <li>If that fails, try to parse {@code subtitleColor} as a decimal string
+	 * without alpha.</li>
+	 * <li>If that also fails, try to parse {@code subtitleColor} as a color
+	 * name (Red, Yellow etc).</li>
+	 * <li>As a last attempt, try to parse {@code subtitleColor} as a system
+	 * property integer.</li>
+	 * </ul>
+	 * If the parsing fails, {@code null} is returned.
+	 *
+	 * @param subtitleColor the {@link String} to attempt to parse a
+	 *            {@link SubtitleColor} from.
+	 * @return A newly created {@link SubtitleColor} or {@code null} it the
+	 *         parsing fails.
+	 */
+	public static SubtitleColor toSubtitleColor(String subtitleColor) {
+		NumberFormatException ne = null;
+		String value = extractHexString(subtitleColor);
+		try {
+			if (value != null) {
+				int r = Integer.parseInt(value.substring(0, 2), 16);
+				int g = Integer.parseInt(value.substring(2, 4), 16);
+				int b = Integer.parseInt(value.substring(4, 6), 16);
+				int a = value.length() == 8 ? Integer.parseInt(value.substring(6, 8), 16) : 0xFF;
+				return new SubtitleColor(r, g, b, a);
+			}
+			value = extractDecString(subtitleColor);
+			if (value != null) {
+				return new SubtitleColor((int) Long.parseLong(value));
+			}
+		} catch (NumberFormatException e) {
+			ne = e;
+		}
+		int[] rgb = getNamedColorValues(subtitleColor);
+		if (rgb != null && rgb.length == 3) {
+			return new SubtitleColor(rgb[0], rgb[1], rgb[2]);
+		}
+		Color color = Color.getColor(subtitleColor);
+		if (color != null) {
+			return new SubtitleColor(color.getRed(), color.getGreen(), color.getBlue(), color.getAlpha());
+		}
+		if (ne != null) {
+			LOGGER.warn("Could not parse subtitle color \"{}\": {}", subtitleColor, ne.getMessage());
+			LOGGER.trace("", ne);
+		} else {
+			LOGGER.warn("Could not parse subtitle color \"{}\"", subtitleColor);
+		}
+		return null;
+	}
+
+	/**
+	 * Returns an array of RGB integer values from a predefined list of color
+	 * names.
+	 *
+	 * @param colorName the name of the color which values to retrieve.
+	 * @return The integer array with RGB values or {@code null} if
+	 *         {@code colorName} isn't among the predefined color names.
+	 */
+	public static int[] getNamedColorValues(String colorName) {
+		colorName = colorName.toUpperCase(Locale.ROOT);
+		if (knownColors.containsKey(colorName)) {
+			return knownColors.get(colorName);
+		}
+		return null;
+	}
+
+	static {
+		// The color values are from https://gist.github.com/XiaoxiaoLi/8031146
```

Comment:
This original reference should be better, IMO:
https://ffmpeg.org/ffmpeg-utils.html#toc-Color